File: 82-remove-duplicates-from-sorted-list-ii/82-remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummy = new ListNode();
        ListNode d = dummy;
        
        while (head!=null) {
            if ( head.next!=null && head.val == head.next.val )
            {
                int val = head.val;
                while ( head!=null && head.val == val )
                    head = head.next;
                continue;
            }
            d.next = head;
            d = d.next;
            head = head.next;
        }
        d.next = head;
               
        return dummy.next;
    }
}

File: 82-remove-duplicates-from-sorted-list-ii/82-remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0,head)
        d = dummy;
        
        while head:
            if head.next and head.next.val == head.val:
                while head.next and head.next.val == head.val:
                    head = head.next
                head = head.next
                d.next = head
            else:
                d = d.next
                head = head.next
            
        
        return dummy.next